packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts

```diff
@@ -83,23 +83,21 @@ describeNoCompat("GC Data Store Aliased", (getTestObjectProvider) => {
         await provider.ensureSynchronized();
     });
 
-    // TODO: fully validate that GC is notified. Currently this tests the race condition
-    // where a remote datastore is summarized before the alias op arrives when trySetAlias is called.
-    // TODO: Remove the itExpects once this issue is fixed https://github.com/microsoft/FluidFramework/issues/8859
-    itExpects("GC is notified when datastores are aliased.",
-    [{ eventName: "fluid:telemetry:ContainerRuntime:GarbageCollector:gcUnknownOutboundReferences" }],
-    async () => {
-        await summarizeOnContainer(container2);
+    it("GC is notified when datastores are aliased.", async () => {
         const containerRuntime1 = mainDataStore1.containerRuntime;
         const aliasableDataStore1 = await containerRuntime1.createDataStore("TestDataObject");
+        const response = await aliasableDataStore1.request({url: "/"});
+        const ds1 = response.value as TestDataObject;
 
         (aliasableDataStore1 as any).fluidDataStoreChannel.bindToContext();
         await provider.ensureSynchronized();
 
         // We run the summary so await this.getInitialSnapshotDetails() is called before the datastore is aliased
-        // and after the datastore is attached. This sets the isRootDataStore. This should be passing as there is
-        // further GC work that will require this test to pass https://github.com/microsoft/FluidFramework/issues/8859
-        await summarizeOnContainer(container2);
+        // and after the datastore is attached. This sets the isRootDataStore to a value.
+        let summaryWithStats = await summarizeOnContainer(container2);
+        let gcState = getGCStateFromSummary(summaryWithStats.summary);
+        assert(gcState?.gcNodes[ds1.handle.absolutePath].unreferencedTimestampMs !== undefined,
+            "AliasableDataStore1 should be unreferenced!");
```

Comment:
nit: Add it should be unreferenced because its not aliased yet and hence not root. This will clearly tell what expectation / issue is if this error is ever thrown in CI loop or telemetry. Similar comment for the assert for referenced.